client/src/main/java/tds/student/Student.java

```diff
@@ -4,37 +4,36 @@
  * Represents a student taking an exam
  */
 public class Student {
-    private long id;
-    private String studentId;
-    private String stateCode;
-    private String clientName;
+    private final long id;
+    private final String externalStudentId;
```

Comment:
I don't think we know that this is the external student id.  I believe where this is being grabbed from they are taking what the student used to login.  Meaning that it could be what they call the external SSID (sometimes Alternate SSID or it could be just the SSID.  This was confusing for us at first.  By default the system is configured (based on the seed data provided) to use the External SSID.  

Not sure if the name SSID would be better, that might confusing too since that is what they call the other SSID (non-external).  Not sure what name should be used, but I think External isn't right since it might not be External SSID.